src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -92,20 +94,46 @@ public function handle(GetResponseEvent $event)
         }
     }
 
-    private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorInterface $guardAuthenticator, GetResponseEvent $event)
+    private function executeGuardAuthenticator($uniqueGuardKey, AuthenticatorInterface $guardAuthenticator, GetResponseEvent $event)
     {
         $request = $event->getRequest();
         try {
             if (null !== $this->logger) {
                 $this->logger->debug('Calling getCredentials() on guard configurator.', array('firewall_key' => $this->providerKey, 'authenticator' => get_class($guardAuthenticator)));
             }
 
+            // abort the execution of the authenticator if it doesn't support the request.
+            if ($guardAuthenticator instanceof GuardAuthenticatorInterface) {
+                // it's a GuardAuthenticatorInterface
+                // we support the previous behaviour to avoid BC break.
```

Comment:
Please replace this comment by `deprecated since version 3.4, to be removed in 4.0`, we basically `grep`on that sentence for deprecating remove deprecated features